File: testdata/precision-suite-ruby/tp_codeinjection.rb
# Code injection: a request-controlled string is passed to eval, executing
# attacker-supplied Ruby — CWE-95 (reported under TAINT-005, the eval/deser
# family). A correct scanner fires TAINT-005.
class CalcController
  def evaluate
    expr = params[:expr]
    result = eval(expr) # nox-expect: TAINT-005
    render plain: result
  end
end

File: testdata/precision-suite-php/tp_codeinjection.php
<?php
// Code injection: a POST field is passed to eval(), so an attacker runs
// arbitrary PHP. CWE-95, TAINT-005.

function compute()
{
    $formula = $_POST['formula'];
    eval("\$result = " . $formula . ";"); // nox-expect: TAINT-005
}
